dashboard/pages/primary_screening/callbacks.py

```diff
@@ -49,14 +49,53 @@ def upload_bmg_data(contents, names, last_modified, stored_uuid, file_storage):
         serialized_processed_df = bmg_df.reset_index().to_parquet()
         file_storage.save_file(f"{stored_uuid}_bmg_df.pq", serialized_processed_df)
 
+    NUM_COLS = 2
```

Comment:
The code for generating file-list component both here and for stage 4 is almost identical and both take a lot of space in here, I strongly recommend to create a function that creates that component for you and place it in the `dashboard.pages.components` module

(sth like:
```python
def make_file_list_component(successfull_filenames: list[str], failed_filenames: list[str], num_cols: int) -> html.Div:
   ...
```